common/attest_plugin.c

```diff
@@ -301,61 +301,83 @@ oe_result_t oe_verify_evidence(
 
     if (!format_id)
     {
-        oe_attestation_header_t* evidence =
-            (oe_attestation_header_t*)evidence_buffer;
-
-        if (evidence_buffer_size < sizeof(oe_attestation_header_t) ||
-            evidence->version != OE_ATTESTATION_HEADER_VERSION)
+        // check whether evidence buffer structure is oe_report
+        oe_report_header_t* report = (oe_report_header_t*)evidence_buffer;
+        // oe_report_header_t size is smaller than oe_attestation_header_t size;
```

Comment:
This assumption may not be true in the future (e.g. oe_attestation_header_t becomes smaller, even though highly unlikely).

To guarantee forward compatibility, I'd suggest combining the 2 if statements. I.e.

if (evidence_buffer_size >= sizeof(oe_report_header_t && 
    report->version == OE_REPORT_HEADER_VERSION)
    // treat as report

Reply:
Make sense. Thanks for pointing out.